### PR TITLE
Auth webhook: use externally provided roles, add collaborators

### PR DESCRIPTION
**Pull request by aermakov-zalando**
Change the webhook to explicitly configure the role mapping, allow collaborator access for test clusters.

### CODE REVIEW

File: cluster/node-pools/master-default/userdata.yaml [hunk 1]
```diff
@@ -246,11 +246,26 @@ write_files:
...
+{{- else }}
+            - --role-mapping=PowerUser=cn=PowerUser,ou={{.Cluster.InfrastructureAccount}}:{{.Cluster.Region}}:{{.Cluster.LocalID}},ou=Kubernetes,ou=apps,dc=zalando,dc=net
+{{- end }}
+            - --role-mapping=ReadOnly=cn=ReadOnly,ou={{.Cluster.InfrastructureAccount}}:{{.Cluster.Region}}:{{.Cluster.LocalID}},ou=Kubernetes,ou=apps,dc=zalando,dc=net
+
           env:
             - name: TEAMS_API_URL
               value: https://teams.auth.zalando.com
-            - name: CLUSTER_ID
-              value: {{if index .Cluster.ConfigItems "webhook_id"}}{{ .Cluster.ConfigItems.webhook_id }}{{else}}{{ .Cluster.ID }}{{end}}
```
Comment (mikkeloscar) [inline cluster/node-pools/master-default/userdata.yaml:253]: Note this was a special case for e2e clusters. We just need to make sure we still allow access. Let's discuss offline :)

Comment (aermakov-zalando) [reply cluster/node-pools/master-default/userdata.yaml:253]: Fixed.

File: cluster/node-pools/master-default/userdata.yaml [hunk 2]
```diff
@@ -246,11 +246,28 @@ write_files:
...
+            - --role-mapping=ReadOnly=cn=ReadOnly,ou=collaborators,ou=Kubernetes,ou=apps,dc=zalando,dc=net
+{{- if eq .Cluster.Environment "test"}}
+            - --role-mapping=PowerUser=cn=Deployer,ou=collaborators,ou=Kubernetes,ou=apps,dc=zalando,dc=net
+{{- else if eq .Cluster.Environment "e2e"}}
+            - --role-mapping=Administrator=cn=Contributor,ou=collaborators,ou=Kubernetes,ou=apps,dc=zalando,dc=net
+{{- end}}
+
+            # Member roles
+{{- if eq .Cluster.Environment "production"}}
+            - --role-mapping=Manual=cn=Manual,ou={{.Cluster.InfrastructureAccount}}:{{.Cluster.Region}}:{{.Cluster.LocalID}},ou=Kubernetes,ou=apps,dc=zalando,dc=net
```
Comment (mikkeloscar) [inline cluster/node-pools/master-default/userdata.yaml]: nit: `{{.Cluster.InfrastructureAccount}}:{{.Cluster.Region}}:{{.Cluster.LocalID}}` could just be `{{.Cluster.ID}}` right? :)

Comment (aermakov-zalando) [reply cluster/node-pools/master-default/userdata.yaml]: Good point.

**Comment by aermakov-zalando** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**